src/app/edit-maprule/disabled-feature/disabled-feature.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, ViewChild, ElementRef } from '@angular/core';
+import { Validators, FormGroup, FormBuilder, FormArray } from '@angular/forms';
+import { EditMapRuleComponent } from '../edit-maprule.component';
+import { FieldConfigService } from 'src/app/core/services/field-config.service';
+
+@Component({
+  exportAs: 'disabled-feature',
+  selector: 'app-disabled-feature',
+  styleUrls: [
+    '../form-table.css',
+    '../preset/preset.component.css',
+    '../../shared/components/content.group.css',
+    './disabled-feature.component.css'
+  ],
+  templateUrl: './disabled-feature.html'
+})
+export class DisabledFeatureComponent {
+
+  constructor(
+    private fb: FormBuilder,
+    private fieldConfig: FieldConfigService,
+    private editMapRule: EditMapRuleComponent
+  ) {}
+
+  @ViewChild("disabledFeature") disabledFeature: ElementRef;
+  _showComponent: boolean = true;
+
+  ngOnInit() { }
+
+  ngAfterViewInit() {
+    this.fieldConfig.emitter.subscribe(
+      (next) => {
+        if (next.type && next.type === 'maprule-init') {
```

Comment:
When we initialize a form with existing maprules data, this event gets fired. This handler checks to see if there are disabled features included and if so makes sure the disabled features template is made to be open.